### PR TITLE
DAOS-14630 doc: Improve MD-on-SSD example config and validate with test

### PR DESCRIPTION
**Pull request by tanabarr**
Ensure example MD-on-SSD config file is up-to-date and valid by adding
a unit test to verify parsing the file results in a valid config.

Skip-func-test: true
Required-githooks: true

### Before requesting gatekeeper:

* [x] Two review approvals and any prior change requests have been resolved.
* [x] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [x] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [x] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [x] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate owners.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

File: utils/config/examples/daos_server_mdonssd.yml [hunk 1]
```diff
@@ -34,21 +50,20 @@ engines:
...
     # wal, meta, data
 
-    # The DAOS server assigns the roles to the bdev tiers specified in the
-    # storage description.  If these default assignments are not suitable,
-    # the roles can be assigned explicitly.  To do this, each of the roles
-    # must be assigned to a bdev tier.  Each tier must have one or more roles
-    # assigned.  The wal and meta roles must be assigned to exactly one tier,
+    # MD-on-SSD roles must be assigned explicitly.  To do this, each of the roles
+    # must be assigned to at least one bdev tier and each tier must have one or more
+    # roles assigned.  The wal and meta roles must be assigned to exactly one tier,
```
Comment (NiuYawei) [inline utils/config/examples/daos_server_mdonssd.yml:55]: I don't think "The wal and meta roles must be assigned to exactly one tier", the only case we don't support is "wal and data are on same tier but meta is on a dedicated tier".

**Review by kjacque — DISMISSED** [review]

File: utils/config/examples/daos_server_mdonssd.yml (hunk 1 above)
Comment (tanabarr) [reply utils/config/examples/daos_server_mdonssd.yml]: fixed

**Comment by tanabarr** [comment]
Change involves only doc change and control unit test updates. Previous CI run completed unit tests and code review related updates, which only involved doc changes, have been applied.

File: utils/config/examples/daos_server_mdonssd.yml [hunk 2]
```diff
@@ -34,21 +50,21 @@ engines:
...
-    # while the data role may be assigned to multiple tiers.
+    # bdev_roles must be specified on each NVMe storage tier to use MD-on-SSD mode
+    # and these roles can only be applied when the first storage tier is of class "ram".
+    # If the first storage tier is of class "dcpm", no bdev_roles can be specified.
+
+    # MD-on-SSD roles must be assigned explicitly.  To do this, each of the roles
+    # must be assigned to at least one bdev tier and each tier must have one or more
+    # roles assigned.  The role configuration where wal and data are on same tier but
+    # meta is on a dedicated tier is not supported.  data is the only role that can be
+    # assigned to multiple bdev tiers.
```
Comment (NiuYawei) [inline utils/config/examples/daos_server_mdonssd.yml:61]: "data is the only role that can be assigned to multiple bdev tiers". As we discussed, this should not be allowed.

**Review by kjacque — APPROVED** [review]

**Review by knard-intel — APPROVED** [review]

**Review by daltonbohning — COMMENTED** [review]
Shouldn't this have run the new test?
`src/control/server/config/server_test.go`

**Comment by tanabarr** [comment]
> Shouldn't this have run the new test? `src/control/server/config/server_test.go`

it did in https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fdaos/detail/PR-13951/1/pipeline no code changes since that run

**Pull request merged**